realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -152,14 +161,25 @@ public boolean isInTransaction() {
      */
     public void addChangeListener(RealmChangeListener listener) {
         checkIfValid();
-        for (WeakReference<RealmChangeListener> ref : changeListeners) {
+        changeListeners.addIfAbsent(listener);
+    }
+
+    /**
+     * For internal use only.
+     * Sometimes we don't know when to unregister listeners (ex: {@link RealmBaseAdapter}), using
+     * a WeakReference the listener.
+     *
+     * @param listener the change listener.
+     */
+    void addChangeListenerAsWeakReference(RealmChangeListener listener) {
```

Comment:
This is a problem as we have a PR for moving all android components to `io.realm.android`. Some alternatives:

1) Make this public...eeew
2) Keep everything in `io.realm`. Also eew, but less so
3) Some other hack, like moving it to `io.realm.internal` and let `BaseRealm` and UI components talk to some API there. If it is considered internal it might make sense? Although having the changelisteners in BaseRealm.java is also nice.